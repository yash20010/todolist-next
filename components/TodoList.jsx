import Task from '../components/Task.jsx'
import styles from '../styles/Home.module.css'

const TodoList = ({ setTask, tasks, setTasks }) => {
  return (
    <>
      {tasks.map((task) => (
        <Task
          key={task._id}
          task={task}
          setTask={setTask}
          tasks={tasks}
          setTasks={setTasks}
        />
      ))}
      {tasks.length === 0 && <h2 className={styles.no_tasks}>No tasks</h2>}
    </>
  )
}

export default TodoList
