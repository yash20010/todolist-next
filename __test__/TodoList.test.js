import React from 'react'
import ReactDOM from 'react-dom'
import TodoList from '../components/TodoList.jsx'
import { render, fireEvent, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<TodoList tasks={[]} />, div)
})

it('renders the todos correctly', () => {
  const tasks = [
    {
      _id: 1,
      task: 'hello',
      completed: false,
      createdAt: new Date(),
    },
    {
      _id: 2,
      task: 'world',
      completed: false,
      createdAt: new Date(),
    },
  ]
  const { getByTestId } = render(<TodoList tasks={tasks} />)
  expect(getByTestId('task_text1')).toHaveTextContent('hello')
  expect(getByTestId('task_text2')).toHaveTextContent('world')
})

it('matches snapshot', () => {
  const tree = renderer.create(<TodoList tasks={[]} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('matches snapshot with two todos', () => {
  const tasks = [
    {
      _id: 1,
      task: 'hello',
      completed: false,
      createdAt: new Date(),
    },
    {
      _id: 2,
      task: 'world',
      completed: false,
      createdAt: new Date(),
    },
  ]
  const tree = renderer.create(<TodoList tasks={tasks} />).toJSON()
  expect(tree).toMatchSnapshot()
})
