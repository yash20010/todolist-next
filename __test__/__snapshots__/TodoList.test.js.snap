// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h2
  className="no_tasks"
>
  No tasks
</h2>
`;

exports[`matches snapshot with two todos 1`] = `
Array [
  <div
    className="task_container"
    data-testid="task-container"
  >
    <input
      checked={false}
      className="check_box"
      data-testid="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <p
      className="task_text"
      data-testid="task_text1"
    >
      hello
    </p>
    <button
      className="edit_task"
      data-testid="edit-btn"
      onClick={[Function]}
    >
      ✎
    </button>
    <button
      className="remove_task"
      data-testid="delete-btn"
      onClick={[Function]}
    >
      ✖
    </button>
  </div>,
  <div
    className="task_container"
    data-testid="task-container"
  >
    <input
      checked={false}
      className="check_box"
      data-testid="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <p
      className="task_text"
      data-testid="task_text2"
    >
      world
    </p>
    <button
      className="edit_task"
      data-testid="edit-btn"
      onClick={[Function]}
    >
      ✎
    </button>
    <button
      className="remove_task"
      data-testid="delete-btn"
      onClick={[Function]}
    >
      ✖
    </button>
  </div>,
]
`;
